simulators/ethereum/graphql/graphql.go

```diff
@@ -156,24 +156,41 @@ func (tc *testCase) run(t *hivesim.T, c *hivesim.Client) {
 		return
 	}
 
+	tc.responseMatch(t, resp.Status, respBytes)
+}
+
+func (tc *testCase) responseMatch(t *hivesim.T, respStatus string, respBytes []byte) error {
 	// Check that the response matches.
 	var got interface{}
 	if err := json.Unmarshal(respBytes, &got); err != nil {
 		t.Fatal("can't decode response:", err)
 	}
-	if !reflect.DeepEqual(tc.gqlTest.Response, got) {
-		prettyQuery, ok := reindentJSON(tc.gqlTest.Request)
-		prettyExpected, _ := json.MarshalIndent(tc.gqlTest.Response, "", "  ")
-		prettyResponse, _ := json.MarshalIndent(got, "", "  ")
-		t.Log("Test failed.")
-		t.Log("HTTP response code:", resp.Status)
-		if ok {
-			t.Log("query:", prettyQuery)
+	// return if a response matches. If not, error out.
+	for i, response := range tc.gqlTest.Responses {
+		if reflect.DeepEqual(response, got) {
+			return nil
+		}
+		if i == len(tc.gqlTest.Responses)-1 {
+			prettyQuery, ok := reindentJSON(tc.gqlTest.Request)
+			prettyResponse, _ := json.MarshalIndent(got, "", "  ")
+			defer func() {
+				t.Log("Test failed.")
+				t.Log("HTTP response code:", respStatus)
+				if ok {
+					t.Log("query:", prettyQuery)
+				}
+				t.Log("expected value(s):")
+				for _, expected := range tc.gqlTest.Responses {
+					prettyExpected, _ := json.MarshalIndent(expected, "", "  ")
+					t.Log(string(prettyExpected), "\n_____________________\n")
+				}
+
+				t.Log("got:", string(prettyResponse))
+				t.Fail()
+			}()
```

Comment:
I think this would be simpler if you first did the for-loop, instead of checking the index and adding a deferred function. Something like this:

```suggestion
	for i, response := range tc.gqlTest.Responses {
		if reflect.DeepEqual(response, got) {
			return nil
		}
	}
	prettyQuery, ok := reindentJSON(tc.gqlTest.Request)
	prettyResponse, _ := json.MarshalIndent(got, "", "  ")
	
	t.Log("Test failed.")
	t.Log("HTTP response code:", respStatus)
	if ok {
		t.Log("query:", prettyQuery)
	}
	t.Log("expected value(s):")
	for _, expected := range tc.gqlTest.Responses {
		prettyExpected, _ := json.MarshalIndent(expected, "", "  ")
		t.Log(string(prettyExpected), "\n_____________________\n")
	}
	t.Log("got:", string(prettyResponse))
	t.Fail()
```